feat: Support brand setting for Universal Login

toshi1127 opened the pull request:
### Proposed Changes

* Support brand setting for Universal Login

Fixes #347, #354 

#### Acceptance Test Output

```
$ make testacc TESTS=TestAccXXX

==> Checking that code complies with gofmt requirements...
?       github.com/alexkappa/terraform-provider-auth0   [no test files]
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_debugDefaults
--- PASS: TestProvider_debugDefaults (0.00s)
=== RUN   TestAccBrand
--- PASS: TestAccBrand (2.11s)
=== RUN   TestAccClientGrant
--- PASS: TestAccClientGrant (29.07s)
=== RUN   TestAccClient
--- PASS: TestAccClient (1.13s)
=== RUN   TestAccClientZeroValueCheck
--- PASS: TestAccClientZeroValueCheck (8.93s)
=== RUN   TestAccClientRotateSecret
--- PASS: TestAccClientRotateSecret (7.91s)
=== RUN   TestAccClientInitiateLoginUri
--- PASS: TestAccClientInitiateLoginUri (0.02s)
=== RUN   TestAccClientJwtScopes
--- PASS: TestAccClientJwtScopes (1.89s)
=== RUN   TestAccClientMobile
--- PASS: TestAccClientMobile (8.32s)
=== RUN   TestAccClientMobileValidationError
--- PASS: TestAccClientMobileValidationError (0.01s)
```

I've confirmed that the colors and logos are set.
![スクリーンショット 2021-03-28 6 48 04](https://user-images.githubusercontent.com/32378535/112735833-a4a15b00-8f91-11eb-9026-75a567249158.png)


<!--- Please keep this note for the community --->

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

<!--- Thank you for keeping this note for the community --->

toshi1127 commented:
I plan to create a document on how to use it once the following questions are resolved and the code review is complete.

1. Should I remove the font setting?
Regarding font customization, the API says that it is supported, but in fact it is supposed to be changed in the Universal Login Page Templates.
(ref: https://community.auth0.com/t/universal-login-custom-font/39217/5)

2. It seems that auth0_tenant can set the color, but should auth0_brand support it as well?

yvovandoorn commented:
Hi @toshi1127, I would remove the font setting considering its undocumented. 

There is going to be a lot of effort in the coming months going into https://auth0.com/docs/universal-login/new-experience/universal-login-page-templates. Long term that is where you will maintain significant changes and the branding fields will be used for situations where the implementor is OK with Auth0's default look and feel of the login box.

toshi1127 commented:
Thank you, @yvovandoorn , for answering my question.

I removed the font and favicon settings, referring to https://auth0.com/docs/universal-login/new-experience/universal-login-page-templatesのAvailable Variables. 
Also added documentation on the auth0_branding module.

kpurdon commented:
This is great! I came looking specifically for https://auth0.com/docs/universal-login/new-experience/universal-login-page-templates. Does it make sense to include the resource for branding templates with this work?

- https://auth0.com/docs/api/management/v2#!/Branding/get_universal_login
- https://auth0.com/docs/api/management/v2#!/Branding/delete_universal_login
- https://auth0.com/docs/api/management/v2#!/Branding/put_universal_login

toshi1127 commented:
@kpurdon , thanks for the feedback.

@yvovandoorn @kpurdon 
It looks like it would be very good to have support for the [Page Templates API ](https://auth0.com/docs/universal-login/new-experience/universal-login-page-templates#page-templates-api).
If there's no problem including it in this PR, I'll implement it!

toshi1127 commented:
Implementation complete!
Please review this PR.

yvovandoorn commented:
Hey @toshi1127, two additions (and undoing my feedback here: https://github.com/alexkappa/terraform-provider-auth0/pull/357#issuecomment-810171264 -- my apologies!). 

Could you re-add the `font` support. It's been added in per https://auth0.com/docs/api/management/v2#!/Branding/patch_branding. 

In addition to `font`, could you also add `favicon_url`?

Once those two are in, I'll get this merged in. Apologies again for going back on my recommendation.

toshi1127 commented:
@yvovandoorn, please don't worry about it.
I can implement the feature to set the font and favicon_url.

Please wait until the implementation is completed.

yvovandoorn commented:
Thanks @toshi1127 (and for implementing these changes!)

toshi1127 commented on auth0/resource_auth0_branding_test.go:
```diff
@@ -0,0 +1,67 @@
+	})
+}
+
+const TestAccBrandingConfigCreate = `
```
template settings for universal login have not been implemented test code.
because it will cause an error if the custom domain is not enabled.

toshi1127 commented:
@yvovandoorn , Implementation is complete, please review it.
